src/google/protobuf/parse_context.h

```diff
@@ -1196,7 +1196,14 @@ template <typename Add, typename SizeCb>
 const char* EpsCopyInputStream::ReadPackedVarint(const char* ptr, Add add,
                                                  SizeCb size_callback) {
   int size = ReadSize(&ptr);
-  size_callback(size);
+  auto element_size = 0;
+  const uint8_t* pData = (uint8_t*)ptr;
+  for (auto i = 0; i < size; i++) {
```

Comment:
this loop is not safe, data corruption (or a truncated input buffer) can cause size to be much larger than pData.